src/anomalib/utils/callbacks/metrics_manager.py

```diff
@@ -0,0 +1,173 @@
+
+        pixel_metric_names: list[str]
+        if self.pixel_metric_names is None:
+            pixel_metric_names = []
+        elif self.task == TaskType.CLASSIFICATION:
+            pixel_metric_names = []
+            logger.warning(
+                "Cannot perform pixel-level evaluation when task type is classification. "
+                "Ignoring the following pixel-level metrics: %s",
+                self.pixel_metric_names,
+            )
+        else:
+            pixel_metric_names = self.pixel_metric_names
+
+        if isinstance(pl_module, AnomalyModule):
+            pl_module.image_metrics = create_metric_collection(image_metric_names, "image_")
+            pl_module.pixel_metrics = create_metric_collection(pixel_metric_names, "pixel_")
+
+            pl_module.image_metrics.set_threshold(pl_module.image_threshold.value)
+            pl_module.pixel_metrics.set_threshold(pl_module.pixel_threshold.value)
+
+    def on_validation_epoch_start(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+    ) -> None:
+        pl_module.image_metrics.reset()
+        pl_module.pixel_metrics.reset()
+
+    def on_validation_batch_end(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+        outputs: STEP_OUTPUT | None,
+        batch: Any,
+        batch_idx: int,
+        dataloader_idx: int = 0,
+    ) -> None:
+        if outputs is not None:
+            self._outputs_to_cpu(outputs)
+            self._collect_output(pl_module.image_metrics, pl_module.pixel_metrics, outputs)
+
+    def on_validation_epoch_end(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+    ) -> None:
+        self._update_metrics_threshold(pl_module)
+        self._log_metrics(pl_module)
+
+    def on_test_epoch_start(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+    ) -> None:
+        pl_module.image_metrics.reset()
+        pl_module.pixel_metrics.reset()
+
+    def on_test_batch_end(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+        outputs: STEP_OUTPUT | None,
+        batch: Any,
+        batch_idx: int,
+        dataloader_idx: int = 0,
+    ) -> None:
+        if outputs is not None:
+            self._outputs_to_cpu(outputs)
+            self._collect_output(pl_module.image_metrics, pl_module.pixel_metrics, outputs)
+
+    def on_test_epoch_end(
+        self,
+        trainer: "trainer.AnomalibTrainer",
+        pl_module: AnomalyModule,
+    ) -> None:
+        self._log_metrics(pl_module)
+
+    def _update_metrics_threshold(self, pl_module: AnomalyModule) -> None:
+        pl_module.image_metrics.set_threshold(pl_module.image_threshold.value.item())
+        pl_module.pixel_metrics.set_threshold(pl_module.pixel_threshold.value.item())
+
+    @staticmethod
+    def _collect_output(
```

Comment:
```suggestion
    def update_metrics(
```

Reply:
Is `_outputs_to_cpu` to compute metrics on cpu? if so can we have a `device` flag in `update_metrics` and set the device when computing the metrics? We might want to compute the metrics on gpu in the future, so this would make it configurable.